datadriven/src/datadriven.rs

```diff
@@ -0,0 +1,217 @@
+use crate::test_data::TestData;
+use crate::test_data_reader::TestDataReader;
+use anyhow::Result;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+/// The main function to run tests
+///
+/// You need to pass the path of `testdata` where store the test cases, and your function
+/// to output expected result.
+///
+/// It will be compare with crate `difference`
+///
+/// It invokes a data-driven test. The test cases are contained in a
+/// separate test file and are dynamically loaded, parsed, and executed by this
+/// testing framework. By convention, test files are typically located in a
+/// sub-directory called "testdata". Each test file has the following format:
+///
+/// ```txt
+/// <command> [arg | arg=val | arg=(val1, val2, ...)]...
```

Comment:
The original version is `<command>[,<command>...]`.

Reply:
I think second command is for the case `subtest`, so I skip this, for confchange and quorum test, they are not neccessary